### PR TITLE
fix(DTFS2-6142): Unissued Tables deadline

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Issue:
* Unissued facilities table showed submission date deadline for MIA

# Changes made:
* Refactored unissued facilities deadline 
* Changed MIA to use now as compared to submission date
* Updated test

### CODE REVIEW

File: gef-ui/server/utils/facility-helpers.js [hunk 1]
```diff
@@ -96,18 +96,36 @@ const areUnissuedFacilitiesPresent = (application) => {
...
 */
-const facilityIssueDeadline = (submissionDate) => {
-  if (submissionDate) {
-  // converts to timestamp from epoch - '+' to convert from str to int
-    const date = new Date(parseInt(submissionDate, 10));
-    const deadlineDate = add(new Date(date), { months: 3 });
+const facilityIssueDeadline = (application) => {
+  // if MIA, then deadline should be 3 months from today
+  if (application.submissionType === CONSTANTS.DEAL_SUBMISSION_TYPE.MIA) {
+    return formatIssueDeadlineDate(new Date());
```
Comment (abhi-markan) [inline gef-ui/server/utils/facility-helpers.js:114]: Argument should be empty, so `new Date()` is then further now as passed as a value to `Date` constructor on LOC:101.
Suggestion: Amend the `formatIssueDealineDate` function to accept null or void argument, if so then use `new Date()` else use `new Date(date)`.

File: gef-ui/server/utils/facility-helpers.test.js [hunk 2]
```diff
@@ -219,17 +221,45 @@ describe('areUnissuedFacilitiesPresent', () => {
...
+    const result = facilityIssueDeadline(application);
+
+    const expected = '08 Feb 2022';
+
+    expect(result).toEqual(expected);
+  });
+
   it('should return null when there is no submissionDate', () => {
-    const result = facilityIssueDeadline();
+    application.submissionType = CONSTANTS.DEAL_SUBMISSION_TYPE.AIN;
```
Comment (abhi-markan) [inline gef-ui/server/utils/facility-helpers.test.js:260]: This test case needs to be re-factored based on above suggestion.

File: gef-ui/server/utils/facility-helpers.test.js [hunk 3]
```diff
@@ -219,17 +221,45 @@ describe('areUnissuedFacilitiesPresent', () => {
 
 describe('facilityIssueDeadline()', () => {
   const timestamp = 1639586124100;
+  const application = {
```
Comment (abhi-markan) [inline gef-ui/server/utils/facility-helpers.test.js:224]: Can we utilise existing Mocks?

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Comments to address please. Thanks.

File: gef-ui/server/utils/facility-helpers.js (hunk 1 above)
Comment (Zainzzkk) [reply gef-ui/server/utils/facility-helpers.js]: done.  Removed many new dates as not required so many times.  Also used || to check if exists or not and either uses date or new Date()

File: gef-ui/server/utils/facility-helpers.test.js (hunk 2 above)
Comment (Zainzzkk) [reply gef-ui/server/utils/facility-helpers.test.js]: added new test

File: gef-ui/server/utils/facility-helpers.js [hunk 4]
```diff
@@ -96,18 +96,38 @@ const areUnissuedFacilitiesPresent = (application) => {
...
 */
-const facilityIssueDeadline = (submissionDate) => {
-  if (submissionDate) {
-  // converts to timestamp from epoch - '+' to convert from str to int
-    const date = new Date(parseInt(submissionDate, 10));
-    const deadlineDate = add(new Date(date), { months: 3 });
+const facilityIssueDeadline = (application) => {
+  // if MIA, then deadline should be 3 months from today
+  if (application.submissionType === CONSTANTS.DEAL_SUBMISSION_TYPE.MIA) {
+    return formatIssueDeadlineDate();
```
Comment (IgnatG) [inline gef-ui/server/utils/facility-helpers.js:116]: can we pass the new Date here? this to keep it consistent with the other calls on line 123 and 130

Comment (Zainzzkk) [reply gef-ui/server/utils/facility-helpers.js:116]: @abhi-markan suggested removing it and keeping the check if exists.  Which should I do?

File: gef-ui/server/utils/facility-helpers.test.js [hunk 5]
```diff
@@ -220,21 +223,66 @@ describe('areUnissuedFacilitiesPresent', () => {
...
+    MOCK_MIN_APPLICATION.manualInclusionNoticeSubmissionDate = 1636373639000;
+
+    const result = facilityIssueDeadline(MOCK_MIN_APPLICATION);
+
+    const expected = '08 Feb 2022';
+
+    expect(result).toEqual(expected);
+  });
+
   it('should return null when there is no submissionDate', () => {
```
Comment (abhi-markan) [inline gef-ui/server/utils/facility-helpers.test.js:256]: Update the description

File: gef-ui/server/utils/facility-helpers.test.js [hunk 6]
```diff
@@ -220,21 +223,66 @@ describe('areUnissuedFacilitiesPresent', () => {
...
-    const result = facilityIssueDeadline();
+    MOCK_AIN_APPLICATION.submissionDate = null;
+    const result = facilityIssueDeadline(MOCK_AIN_APPLICATION);
 
     expect(result).toEqual(null);
   });
 });
 
+describe('formatIssueDeadlineDate()', () => {
+  const timestamp = 1639586124100;
```
Comment (abhi-markan) [inline gef-ui/server/utils/facility-helpers.test.js:265]: Best to apply it to `MOCK_AIN_APPLICATION.submissionDate` in local scope.

File: gef-ui/server/utils/facility-helpers.test.js [hunk 7]
```diff
@@ -220,21 +223,66 @@ describe('areUnissuedFacilitiesPresent', () => {
...
+
+  it('should return a correct timestamp 3 months in advance in the right format from date if providing date', () => {
+    const date = new Date(parseInt(timestamp, 10));
+    const result = formatIssueDeadlineDate(date);
```
Comment (abhi-markan) [inline gef-ui/server/utils/facility-helpers.test.js:269]: Pass in `MOCK_AIN_APPLICATION.submissionDate`.

File: gef-ui/server/utils/facility-helpers.test.js [hunk 8]
```diff
@@ -220,21 +223,66 @@ describe('areUnissuedFacilitiesPresent', () => {
...
+
+    const expected = '15 Mar 2022';
+
+    expect(result).toEqual(expected);
+  });
+
+  it('should return a correct timestamp 3 months in advance from today if do not provide a date', () => {
+    const result = formatIssueDeadlineDate();
```
Comment (abhi-markan) [inline gef-ui/server/utils/facility-helpers.test.js:277]: Pass in `MOCK_AIN_APPLICATION.submissionDate`

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Good work, minor comments to address.

File: gef-ui/server/utils/facility-helpers.test.js (hunk 8 above)
Comment (Zainzzkk) [reply gef-ui/server/utils/facility-helpers.test.js:277]: This is meant to be blank though?

File: gef-ui/server/utils/facility-helpers.test.js (hunk 6 above)
Comment (Zainzzkk) [reply gef-ui/server/utils/facility-helpers.test.js]: discussed

File: gef-ui/server/utils/facility-helpers.test.js (hunk 7 above)
Comment (Zainzzkk) [reply gef-ui/server/utils/facility-helpers.test.js:267]: discussed

File: gef-ui/server/utils/facility-helpers.test.js (hunk 5 above)
Comment (Zainzzkk) [reply gef-ui/server/utils/facility-helpers.test.js:254]: discussed

**Pull request merged**